Use `pint.get_application_registry` to improve interoperability with other modules (#3357)
hyperspy/hyperspy: doc/user_guide/pint_unit_registry.rst

.. _pint_unit_registry:

> Operation with Pint Quantity
> ****************************

Reviewer: I would like the word unit to appear in the TOC and title, e.g.:

```suggestion
Unit Handling with Pint Quantity
********************************
```

HyperSpy uses the `pint <https://pint.readthedocs.io>`_ library to handle unit conversion.

> To be interoperatable with other modules, hyperspy uses the default pint :class:`pint.UnitRegistry` 

Reviewer: ```suggestion
To be interoperatable with other modules, HyperSpy uses the default pint :class:`pint.UnitRegistry` 
```

provided by the :func:`pint.get_application_registry` function as described in the sections
`having a shared registry <https://pint.readthedocs.io/en/stable/getting/pint-in-your-projects.html>`_

> and the `serialization <https://pint.readthedocs.io/en/stable/advanced/serialization.html>`_

Reviewer: ```suggestion
and `serialization <https://pint.readthedocs.io/en/stable/advanced/serialization.html>`_
```

of the pint user guide.

> For example, to use pint quantity object from :class:`~.axes.UniformDataAxis`, the same
> unit registry needs to be used:

Reviewer: ```suggestion
For example, in the case of the  ``scale_as_quantify``  pint quantity object from :class:`~.axes.UniformDataAxis`, the default pint :class:`pint.UnitRegistry` is used:
```

.. code-block:: python

    >>> s = hs.signals.Signal1D(np.arange(10))
    >>> s.axes_manager[0].scale_as_quantity
    <Quantity(1.0, 'dimensionless')>
    >>> s.axes_manager[0].scale_as_quantity = '2.5 µm'
    >>> s.axes_manager[0].scale_as_quantity
    <Quantity(2.5, 'micrometer')>

> Use :func:`pint.get_application_registry` to get pint default unit registry:

Reviewer: ```suggestion
Then, use :func:`pint.get_application_registry` to get the handle of the same instance of :class:`pint.UnitRegistry` used by HyperSpy and therefore necessary to perform operation on this pint quantity:
```

.. _pint_unit_registry:

Unit Handling with Pint Quantity
********************************

HyperSpy uses the `pint <https://pint.readthedocs.io>`_ library to handle unit conversion.
To be interoperatable with other modules, HyperSpy uses the default pint :class:`pint.UnitRegistry` 
provided by the :func:`pint.get_application_registry` function as described in the sections
`having a shared registry <https://pint.readthedocs.io/en/stable/getting/pint-in-your-projects.html>`_
and `serialization <https://pint.readthedocs.io/en/stable/advanced/serialization.html>`_
of the pint user guide.

For example, in the case of the  ``scale_as_quantify``  pint quantity object from :class:`~.axes.UniformDataAxis`,
the default pint :class:`pint.UnitRegistry` is used:

.. code-block:: python

    >>> s = hs.signals.Signal1D(np.arange(10))
    >>> s.axes_manager[0].scale_as_quantity
    <Quantity(1.0, 'dimensionless')>
    >>> s.axes_manager[0].scale_as_quantity = '2.5 µm'
    >>> s.axes_manager[0].scale_as_quantity
    <Quantity(2.5, 'micrometer')>

Then, using :func:`pint.get_application_registry` get the handle of the same instance of :class:`pint.UnitRegistry`

> used by HyperSpy and used it to operate on this pint quantity:

Reviewer: ```suggestion
used by HyperSpy and use it to operate on this pint quantity:
```

    >>> s.axes_manager[0].scale_as_quantity
    <Quantity(2.5, 'micrometer')>

> Then, use :func:`pint.get_application_registry` to get the handle of the same instance of :class:`pint.UnitRegistry` used by HyperSpy and therefore necessary to perform operation on this pint quantity:

Reviewer: ```suggestion
Then, use :func:`pint.get_application_registry` to get the handle of the same instance of :class:`pint.UnitRegistry`
used by HyperSpy, necessary to operate on this pint quantity:
```
